### PR TITLE
[python-package] migrate test_sklearn.py to pytest

### PR DESCRIPTION
**Pull request by thomasjpfan**
Closes https://github.com/microsoft/LightGBM/issues/3732

### CODE REVIEW

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for all your help with these tests! I compared with https://www.textcompare.org/index.html?id=600e2bdfcc09e90017e5a111 and all the changes look good.

Can you please also remove `import unittest` on line 6?

File: tests/python_package_test/test_sklearn.py [hunk 1]
```diff
@@ -80,1058 +80,1072 @@ def multi_logloss(y_true, y_pred):
...
+            for metric_name in metric_names:
+                assert metric_name in gbm.evals_result_[eval_set_name]
+
+                actual = len(gbm.evals_result_[eval_set_name][metric_name])
+                expected = assumed_iteration + (params_fit['early_stopping_rounds']
+                                                if eval_set_name != 'training'
+                                                and assumed_iteration != gbm.n_estimators else 0)
+                assert expected == actual
+                assert (assumed_iteration
+                        if eval_set_name != 'training' else gbm.n_estimators == gbm.best_iteration_)
```
Comment (StrikerRUS) [inline tests/python_package_test/test_sklearn.py:1001]: Please make this assert more readable by splitting into two asserts.
```suggestion
                if eval_set_name != 'training':
                    assert assumed_iteration == gbm.best_iteration_
                else:
                    assert gbm.n_estimators == gbm.best_iteration_
```

**Review by StrikerRUS — APPROVED** [review]
Perfect!

**Review by jameslamb — APPROVED** [review]
Thanks so much for all the help with this issue! We hope you'll contribute more to LightGBM in the future.

**Pull request merged**